build-logic/module-identity/src/main/kotlin/gradlebuild/identity/extension/ModuleIdentityExtension.kt

```diff
@@ -35,7 +35,19 @@ abstract class ModuleIdentityExtension(val tasks: TaskContainer, val objects: Ob
     abstract val snapshot: Property<Boolean>
     abstract val promotionBuild: Property<Boolean>
 
+    /**
+     * The actual build branch.
+     */
     abstract val gradleBuildBranch: Property<String>
+
+    /**
+     * The base branch for pre-tested commit, i.e. the branch is not `master`/`release`,
+     * but it's tested in the same way of `master`/`release`.
+     *
+     * The pre-tested commit branch name looks like "pre-test/master/queue/alice/personal-branch".
+     */
+    abstract val preTestedCommitBaseBranch: Property<String>
```

Comment:
I think this should actually only be a `Provider`, not a property. It is always derived from the `gradleBuildBranch`, so its value is gradleBuildBranch.map { toPreTestedCommitBaseBranch(it) }. There is no need to make this a mutable property.